contracts/BitGuildFeeOracle.sol

```diff
@@ -0,0 +1,53 @@
+pragma solidity ^0.4.24;
+
+import "./BitGuildAccessAdmin.sol";
+
+
+/**
+ * @title BitGuildFeeOracle
+ * @dev Fee definition
+ */
+contract BitGuildFeeOracle is BitGuildAccessAdmin {
+    // @dev fixed is not supported. use percent * 100 to include 2 decimals
+    uint public percentFee = 300; // Fees bitguild marketplace charges.
+
+    event FeeChanged(uint newPercentFee, uint oldPercentFee, address operator);
+
+    /**
+     * @dev Allow operators to update the fee
+     * @param _newPercentFee New fee in percent x 100
+     */
+    function updateFee(uint _newPercentFee) public onlyOperator {
+        require(
+            _newPercentFee > 0 && _newPercentFee < 10000,
+            "Invalid percent fee."
+        );
+
+        emit FeeChanged(_newPercentFee, percentFee, msg.sender);
+        percentFee = _newPercentFee;
+    }
+
+    /**
+     * @dev Calculate the fee
+     * @param _price Base price for fee calculation
+     */
+    function getFee(uint _price) public view returns(uint fee) {
+        require(
+            _price > 0,
+            "Invalid price."
+        );
+
+        fee = _safeMul(_price, percentFee) / 10000; // adjust for percent and decimal. division always truncate
```

Comment:
I also don't think you generally want to return fee as a part of the price. Let the marketplace do that. Otherwise you will have to do the reverse calculations in the UI. So better to return as a percentage.

Reply:
Or actually maybe as a tuple. So you can do that math in one place and then other contract and the UI just to use the result. To avoid rounding errors between different stacks.